clients/src/main/java/org/apache/kafka/clients/consumer/internals/CloseOptionsInternal.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer.internals;
+
+import org.apache.kafka.clients.consumer.CloseOptions;
+
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * This class represents an internal version of {@link CloseOptions}, used for internal processing of consumer shutdown options.
+ *
+ * <p>While {@link CloseOptions} is the user-facing class, {@code CloseOptionsInternal} is intended for accessing internal fields
+ * and performing operations within the Kafka codebase. This class should not be used directly by users.
+ * It extends {@link CloseOptions} and provides getters for the {@link GroupMembershipOperation} and timeout values.</p>
+ */
+public class CloseOptionsInternal extends CloseOptions {
```

Comment:
Sorry if I'm behind on the discussion and missed something, but IIRC we ended up agreeing to get rid of the `CloseOptionsInternal` class since there's no internal implementation stuff to hide. We can just move the getters to the public `CloseOptions` class (which are nice to have anyways)

Reply:
I will fix this in the next commit.